packages/flutter/lib/src/material/toggle_buttons.dart

```diff
@@ -668,33 +669,69 @@ class ToggleButtons extends StatelessWidget {
       final BorderSide borderSide = _getBorderSide(index, theme, toggleButtonsTheme);
       final BorderSide trailingBorderSide = _getTrailingBorderSide(index, theme, toggleButtonsTheme);
 
-      return _ToggleButton(
-        selected: isSelected[index],
-        textStyle: textStyle,
-        constraints: constraints,
-        color: color,
-        selectedColor: selectedColor,
-        disabledColor: disabledColor,
-        fillColor: fillColor,
-        focusColor: focusColor ?? toggleButtonsTheme.focusColor,
-        highlightColor: highlightColor ?? toggleButtonsTheme.highlightColor,
-        hoverColor: hoverColor ?? toggleButtonsTheme.hoverColor,
-        splashColor: splashColor ?? toggleButtonsTheme.splashColor,
-        focusNode: focusNodes != null ? focusNodes![index] : null,
-        onPressed: onPressed != null
-          ? () {onPressed!(index);}
-          : null,
-        mouseCursor: mouseCursor,
+      final TextStyle currentTextStyle = textStyle
+        ?? toggleButtonsTheme.textStyle
+        ?? theme.textTheme.bodyText2!;
+      final BoxConstraints currentConstraints = constraints
+        ?? toggleButtonsTheme.constraints
+        ?? const BoxConstraints(
+             minWidth: kMinInteractiveDimension,
+             minHeight: kMinInteractiveDimension,
+           );
+
+      return _SelectToggleButton(
         leadingBorderSide: leadingBorderSide,
         borderSide: borderSide,
         trailingBorderSide: trailingBorderSide,
         borderRadius: edgeBorderRadius,
-        clipRadius: clipBorderRadius,
         isFirstButton: index == 0,
         isLastButton: index == children.length - 1,
         direction: direction,
         verticalDirection: verticalDirection,
-        child: children[index],
+        child: ClipRRect(
+          borderRadius: clipBorderRadius,
+          child: TextButton(
+            focusNode: focusNodes != null ? focusNodes![index] : null,
+            style: ButtonStyle(
+              backgroundColor: _BackgroundColor(
+                onPressed != null && isSelected[index],
```

Comment:
If you define a isSelected variable above, then these constructors and their positional parameters should fit neatly on one line. The same `onPressed != null && isSelected[index]` appears in the overlayColor value too; definitely wouldn't hurt to factor it out.

Reply:
Great idea, I added a local variable and made those into a single line 

Also added regression for https://github.com/flutter/flutter/issues/97302 (I added it earlier too but somehow missed it during buttonstyle revert).  This test would be able to close the issue :)